test/parallel/test-runner-run.mjs

```diff
@@ -549,6 +553,19 @@ describe('require(\'node:test\').run', { concurrency: true }, () => {
       // eslint-disable-next-line no-unused-vars
       for await (const _ of stream);
     });
+
+    it('should only allow a boolean in options.bail', async () => {
+      [Symbol(), {}, [], () => { }, 0, 1, 0n, 1n, '', '1', Promise.resolve([])]
+        .forEach((bail) => assert.throws(() => run({ bail }), {
+          code: 'ERR_INVALID_ARG_TYPE'
```

Comment:
Just out of curiosity, are we sure that these errors actually correspond to the use of `bail`? For example, it's possible that `ERR_INVALID_ARG_TYPE` or `ERR_INVALID_ARG_VALUE` are thrown by `run()` completely unrelated to the `bail` option.

Reply:
I've written the test following a red-green cycle.
I'll take another look and I'll add a comment here 😁

Reply:
Another option is to update the assertion to check the code and the message.

Reply:
I agree, we should probably do the same for the other "validation" tests as well

Reply:
done